zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderUnitTest.java

```diff
@@ -65,4 +69,314 @@ public void shouldBeInitializedWithDoubleCapacityOfGivenArray() {
         // then
         assertThat(capacity, is(2 * TEST_ARRAY.length));
     }
+
+    @Test
+    public void shouldIncreaseCapacityFrom5To20() {
+        // given
+        byteBuilder = new ByteBuilder(5);
+        // when
+        byteBuilder.ensureCapacity(20);
+        int capacity = byteBuilder.capacity();
+        // then
+        assertThat(capacity, is(20));
+    }
+
+    @Test
+    public void shouldHaveASizeOf0() {
```

Comment:
It would be better with e.g. `shouldHaveZeroSizeByDefault`, otherwise it's not clear what/when behaviour is expected, specially with the following method (i.e. should have a size of 0 or 3?).

Reply:
Test name changed